### PR TITLE
cpplint and other code cleanups

### PR DESCRIPTION
**Pull request by mathieucarbou**

### CODE REVIEW

File: .github/workflows/cpplint.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+name: cpplint
```
Comment (mathieucarbou) [inline .github/workflows/cpplint.yml:1]: if you activate Github Actions on this repo (free), will run a cpplint for each push and PR.

File: library.json [hunk 2]
```diff
@@ -6,14 +6,15 @@
...
-        }
+                }
     ],
     "repository": {
         "type": "git",
          "url": "https://github.com/rwmingis/InterruptButton"
     },
     "version": "1.0.0",
     "frameworks": "Arduino",
+    "platforms": ["espressif32"],
```
Comment (mathieucarbou) [inline library.json:17]: I am using strict verification of libraries in platformio:

```
lib_compat_mode = strict
lib_ldf_mode = deep+
```

So this change will allow the library to be accepted where it should be.

**Comment by mathieucarbou** [comment]
and thanks for facd8a2203e848af6b0d26bcabdf29fb4d324b5a by the way ;-)

**Comment by mathieucarbou** [comment]
Note: tried to get rid of the `-runtime/references` filter, but the different usages of `esp_timer_handle_t &timer` makes it complicated.

**Review by rwmingis — APPROVED** [review]

**Comment by rwmingis** [comment]
Thanks Mathieu.  All good suggestions.  I will admit, navigating github is not one of my strong points, so I have manually implemented your points directly into the repository files in main branch.  Cheers.

**Pull request closed without merging**